lib/core/MantraData.js

```diff
@@ -0,0 +1,303 @@
+     * @returns {boolean} - True if the relationship name exists in the schema property.
+     */
+    _isRelationship(name) {
+        const relationships = this.schema.relationships;
+        const HAS_RELATIONSHIPS = (relationships !== undefined);
+        return HAS_RELATIONSHIPS && relationships.some((rel) => rel.name === name);;
+    };
+
+    /**
+     * Re-sets the name, schema and id property.
+     * Appends the parentEntity object to the parent property.
+     * Updates the path property and trimms the _path property.
+     * @param {string} name - The relationship name.
+     */
+    _setRelationship(name) {
+        const parentEntity = { schema: this.schema, name: this.name, id: this.id };
+        this._addParent(parentEntity);
+        this._setName(name);
+        this._setRelationshipSchema(name);
+        this._resetId();
+        this.updatePath(name);
+    };
+
+    /**
+     * Set the schema property to the returned relationship.
+     * @param {string} name - The relationship name.
+     */
+    _setRelationshipSchema(name) {
+        const relationships = this.schema.relationships;
+        this.schema = relationships.find((rel) => rel.name === name);
+    };
+
+    /**
+     * Verify if the value parameter is a valid id.
+     * @param {string} value - The id to be validated.
+     * @returns {boolean} - True if the relationship name exists in the schema property.
+     */
+    _isId(value) {
+        return IS_ID_NUMBER(value);
+    };
+
+    /**
+     * Set the id property to the id parameter.
+     * Updates the path property and trimms the _path property.
+     * @param {string} id - The id value.
+     */
+    _setId(id) {
+        this.id = id;
+        this._trimPath(id);
+    };
+
+    /**
+     * Sets the id property to it's initial value.
+     */
+    _resetId() {
+        this.id = '';
+    };
+
+    /**
+     * Append the previous entity object to the parents property.
+     * @param {string} entity - The previous entity object.
+     */
+    _addParent(entity) {
+        this.parents.push(entity);
+    };
+
+    /**
+     * Sets the action property to the returned action based on the _path property.
+     */
+    setAction() {
+        this.action = this._getAction();
+    };
+
+    /**
+     * Gets a custom action or a default action object based on the _path property.
+     * @returns {object} - A default or custom action object.
+     */
+    _getAction() {
+        if (this._isCustomAction()) 
+            return this._getCustomAction();
+        else 
+            return this._getDefaultAction();
+    };
+
+    /**
+     * Verify if the first string in the _path parameter is a custom action.
+     * @returns {boolean} - True if a custom action matches the action name.
+     */
+    _isCustomAction() {
+        const [action = ''] = this._path.split('.');
+        const HAS_ACTIONS = this.schema.actions;
+        return HAS_ACTIONS && this.schema.actions.hasOwnProperty(action);
```

Comment:
Create every needed Boolean evaluation as constant

Reply:
This case is similar to the one in the "_isRelationship" method because if I access the "hasOwnProperty" when "actions" is not a valid object then an error will be throw in the application.

It doesn't mean I can't apply the changes but we need to be aware that I could've happened. Also, since we have try/catch statements in the "_getConfig" method of the MantraForm class then these errors will be resolved by a console.error().

The code modification will look like:
```suggestion
        const IS_ACTION = this.schema.actions.hasOwnProperty(action);
        return HAS_ACTIONS && IS_ACTION;
```

Reply:
Yeah bro my bad, let's leave the javascript boolean evaluation as it is, it is clearer! Thanks!